core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1261,8 +1260,6 @@ protected Scope visitTable(Table table, Optional<Scope> scope)
                         Optional.of(name),
                         Optional.of(column.getName()),
                         false);
-                ColumnHandle columnHandle = columnHandles.get(column.getName());
-                checkArgument(columnHandle != null, "Unknown field %s", field);
```

Comment:
@kokosing @Praveen2112 I think this check is redundant. Wdyt?

Reply:
It is redundant. It should never happen as otherwise we should raise `TrinoException` here. It is just defensive programming. You could replace it with `requireNonNull`.

Reply:
Thank you. I'll just remove this lines.

Reply:
Please don't remove it.